File: app/models/dummy/DummyOrder.java
package models.dummy;

import models.*;

import java.util.ArrayList;
import java.util.Date;
import java.util.List;

/**
 * Created by Beemen on 31/07/2015.
 */
public class DummyOrder implements IOrder {

    String _OrderUniqueID;
    @Override
    public String OrderUniqueID() {
        return _OrderUniqueID;
    }

    @Override
    public void OrderUniqueID(String value) {
        _OrderUniqueID = value;
    }

    @Override
    public String OrderTitle() {
        return null;
    }

    @Override
    public void OrderTitle(String value) {

    }

    private final List<OrderArchiveReference> _Archives = new ArrayList<>();
    @Override
    public List<OrderArchiveReference> Archives() {
        return _Archives;
    }

    @Override
    public List<DeliveryMethod> OrderOrigins() {
        return null;
    }

    @Override
    public List<String> EndUserNotes() {
        return null;
    }

    @Override
    public List<String> ArchivistNotes() {
        return null;
    }

    private Date _IssueDate;
    @Override
    public Date IssueDate() {
        return _IssueDate;
    }

    @Override
    public void IssueDate(Date value) {
        _IssueDate = value;
    }

    IEndUser _User;
    @Override
    public IEndUser User() {
        return _User;
    }

    @Override
    public void User(IEndUser value) {
        _User = value;
    }

    @Override
    public Date PlannedDate() {
        return null;
    }

    @Override
    public void PlannedDate(Date value) {

    }

    @Override
    public DateText ExpectedReadyDate() {
        return null;
    }

    @Override
    public void ExpectedReadyDate(DateText value) {

    }

    @Override
    public IArchivist Archivist() {
        return null;
    }

    @Override
    public void Archivist(IArchivist value) {

    }

    @Override
    public List<String> InternalNotes() {
        return null;
    }

    @Override
    public List<DeliveryFormat> DeliveryFormats() {
        return null;
    }

    @Override
    public Boolean Add(IArchive archive) {
        return Add(archive, null);
    }

    @Override
    public Boolean Add(IArchive archive, IDissemination dissemination) {
        this.Archives().add(
                new OrderArchiveReference(){{
                    Archive = archive;
                    Dissemination = dissemination;
                    Status =new OrderStatus(){{
                        Status = "";
                        StatusDate = new Date();
                    }};
                }}
        );
        return true;
    }

    @Override
    public Boolean Remove(IArchive archive) {
        for (OrderArchiveReference orderArchiveReference:this.Archives()){
            if(orderArchiveReference.Archive.ReferenceCode().equals(archive.ReferenceCode())){
                Archives().remove(orderArchiveReference);
                return true;
            }
        }
        return false;
    }
}

File: Models/dummy/DummyOrder.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;

namespace end_user_gui.Models.dummy
{
    public class DummyOrder : IOrder
    {
        public String OrderUniqueID { get; set; }
        public String OrderTitle { get; set; }

        private List<OrderArchiveReference> _Archives = new List<OrderArchiveReference>();
        public List<OrderArchiveReference> Archives { get { return _Archives; } }

        private List<DeliveryMethod> _OrderOrigins = new List<DeliveryMethod>();
        public List<DeliveryMethod> OrderOrigins { get { return _OrderOrigins; } }

        private List<String> _EndUserNotes = new List<string>();
        public List<String> EndUserNotes { get { return _EndUserNotes; } }

        private List<String> _ArchivistNotes = new List<string>();
        public List<String> ArchivistNotes { get { return _ArchivistNotes; } }

        public DateTime IssueDate { get; set; }
        public EndUser User { get; set; }
        public DateTime PlannedDate { get; set; }
        public DateText ExpectedReadyDate { get; set; }
        public Archivist Archivist { get; set; }

        private List<String> _InternalNotes = new List<string>();
        public List<String> InternalNotes { get { return _InternalNotes; } }

        private List<DeliveryFormat> _DeliveryFormats = new List<DeliveryFormat>();
        public List<DeliveryFormat> DeliveryFormats { get { return _DeliveryFormats; } }

        public Boolean Add(Archive archive)
        {
            return Add(archive, null);
        }

        public Boolean Add(Archive archive, Dissemination dissemination)
        {
            this.Archives.Add(
                    new OrderArchiveReference()
                    {
                        Archive = archive,
                        Dissemination = dissemination,
                        Status = new OrderStatus()
                        {
                            Status = "",
                            StatusDate = new DateTime(),
                        }
                    }
            );
            return true;
        }

        public Boolean Remove(Archive archive)
        {
            // TODO: LINQ
            foreach (OrderArchiveReference orderArchiveReference in this.Archives)
            {
                if (orderArchiveReference.Archive.ReferenceCode.Equals(archive.ReferenceCode))
                {
                    Archives.Remove(orderArchiveReference);
                    return true;
                }
            }
            return false;
        }
    }
}